internal/runners/initialize/init.go

```diff
@@ -239,60 +238,83 @@ func (r *Initialize) Run(params *RunParams) (rerr error) {
 	}
 
-	if version == "" {
-		return nil // nothing to verify
+	if len(pkgs) == 0 {
+		return nil, locale.NewInputError("err_init_language_not_found", "The selected language cannot be found")
 	}
 
-	pkgs, err := model.SearchIngredientsStrict(model.NewNamespaceLanguage(), lang.Requirement(), false, true)
-	if err != nil {
-		return locale.WrapError(err, "err_init_verify_language", "Inventory search failed unexpectedly")
+	knownVersions := make([]string, len(pkgs))
+	for i, pkg := range pkgs {
+		knownVersions[i] = pkg.Version
 	}
+	return knownVersions, nil
+}
 
-	if len(pkgs) == 0 {
-		return locale.NewInputError("err_init_language_not_found", "The selected language cannot be found")
+// Can be overridden in tests.
+var getKnownVersions func(language.Language) ([]string, error) = getKnownVersionsFromPlatform
+
+func deriveVersion(lang language.Language, version string) (string, error) {
+	err := lang.Validate()
+	if err != nil {
+		return "", errs.Wrap(err, "Failed to validate language")
 	}
 
-	for _, pkg := range pkgs {
-		if strings.HasPrefix(pkg.Version, version) {
-			return nil
+	if version == "" {
+		// Return default language.
+		langs, err := model.FetchSupportedLanguages(model.HostPlatform)
+		if err != nil {
+			multilog.Error("Failed to fetch supported languages (using hardcoded default version): %s", errs.JoinMessage(err))
+			return lang.RecommendedVersion(), nil
 		}
-	}
 
-	return errs.AddTips(
-		locale.NewInputError(
-			"err_init_language_version_not_found",
-			"The selected version of the language cannot be found",
-		),
-		locale.Tl(
-			"version_not_found_check_format",
-			"Please ensure that the version format is valid.",
-		),
-	)
-}
+		for _, l := range langs {
+			if lang.String() == l.Name || (lang == language.Python3 && l.Name == language.Python3.Requirement()) {
+				return l.DefaultVersion, nil
+			}
+		}
 
-func deriveVersion(lang language.Language, version string) string {
-	if version != "" {
-		return version
+		multilog.Error("Could not find requested language in fetched languages (using hardcoded default version): %s", lang)
+		return lang.RecommendedVersion(), nil
 	}
 
-	langs, err := model.FetchSupportedLanguages(model.HostPlatform)
+	// Fetch known list of languages and verify the given version matches it, either exactly or partially.
+	knownVersions, err := getKnownVersions(lang)
 	if err != nil {
-		multilog.Error("Failed to fetch supported languages (using hardcoded default version): %s", errs.JoinMessage(err))
-		return lang.RecommendedVersion()
+		return "", errs.Wrap(err, "Unable to get known versions for language %s", lang.Requirement())
 	}
 
-	for _, l := range langs {
-		if lang.String() == l.Name || (lang == language.Python3 && l.Name == language.Python3.Requirement()) {
-			return l.DefaultVersion
+	prefix := strings.Replace(strings.Replace(version, ".x", "", 1), ".X", "", 1) // strip any wildcard
```

Comment:
We're just verifying the wildcarded version exists in the first place. e.g. 3.10.x will check that 3.10 is a valid version prefix.